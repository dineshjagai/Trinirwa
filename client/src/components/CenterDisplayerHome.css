.dropZoneOverlay,
.FileUpload {
    width: 283px;
    height: 71px;
}

.dropZoneOverlay {
    border: dotted 1px;
    font-family: cursive;
    color: #7066fb;
    position: absolute;
    top: 0px;
    text-align: center;
}

.FileUpload {
    opacity: 0;
    position: relative;
    z-index: 1;
}

.whatIsOn {
    position: sticky;
    position: -webkit-sticky;
    top: 0;
}

.tweet_items {
    overflow: auto;
}

.tweetIpt {
    position: sticky;
    position: -webkit-sticky;
    top: 0;
}

#hashTag {
    background-color: #D0EAE4;
    border-radius: 8px;
}

.filterBox {
    background-color: #0C8367;
    border-radius: 8px;
    display: flex;
    justify-content: center;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);

}