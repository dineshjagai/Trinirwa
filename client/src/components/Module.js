import axios from 'axios';

export function getUserInformation(username) {
  const prm = axios({
    method: 'GET',
    url: `/api/${username}`,
  });
  return prm;
}

export function addInterest(newInterest, id) {
  const addUrl = `/profile/interest/${id}`;
  return axios({
    method: 'POST',
    url: addUrl,
    data: {
      interest: newInterest,
    },
  });
}

export function deleteInterest(interestToDelete, id) {
  const delUrl = `/profile/delete/interest/${id}`;
  return axios({
    method: 'DELETE',
    url: delUrl,
    data: {
      interest: interestToDelete,
    },
  });
}

export function fetchTweets(id) {
  const getUrl = `/profile/tweet/${id}`;
  return axios({
    method: 'GET',
    url: getUrl,
  });
}

export function getFollowers(id) {
  const getUrl = `/profile/followers/${id}`;
  return axios({
    method: 'GET',
    url: getUrl,
  });
}

export function deleteProfile(id) {
  const delUrl = `/profile/delete/${id}`;
  return axios({
    method: 'DELETE',
    url: delUrl,
  });
}

export const addUser = (usernameReg, passwordOneReg, firstName, lastName, email) => {
  const prm = axios({
    method: 'POST',
    url: '/register',
    data: {
      username: usernameReg,
      password: passwordOneReg,
      first_name: firstName,
      last_name: lastName,
      email,
    },
  });
  return prm;
};

export const getUid = (usernameReg) => {
  const prm = axios({
    method: 'POST',
    url: '/userUid',
    data: {
      username: usernameReg,
    },
  });
  return prm;
};

export const userLogin = (usernameReg, password) => {
  const prm = axios({
    method: 'POST',
    url: '/login',
    data: {
      username: usernameReg,
      password,
    },
  });
  return prm;
};
