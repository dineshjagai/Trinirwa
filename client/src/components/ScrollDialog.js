import React from 'react';
import './ScrollDialog.css';

export default function ScrollDialog() {
  const handleClickOpen = () => {
    document.getElementById('MyModal').setAttribute('display', 'none');
  };
  return (
    <div>
      <button className="btn" id="delete" style={{ color: '#00695c', border: 'none', backgroundColor: 'inherit' }} onClick={() => handleClickOpen()} type="button"> All followers</button>
      <div id="myModal" className="modal">
        <div className="modal-content">
          <span className="close">&times;</span>
          <p>Some text in the Modal..</p>
        </div>
      </div>
    </div>
  );
}
