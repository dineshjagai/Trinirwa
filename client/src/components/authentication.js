export default function isValidPassword(password) {
  const strongRegex = new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,})');
  return strongRegex.test(password);
}

export function isValidUsername(userName) {
  const valid = new RegExp('^[a-zA-Z0-9_]*$');
  return valid.test(userName);
}

export function isValidEmail(email) {
  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(String(email).toLowerCase());
}
