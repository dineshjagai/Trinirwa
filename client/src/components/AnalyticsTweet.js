import React, { useEffect, useState } from 'react';
import { withStyles } from '@material-ui/core/styles';
import Dialog from '@material-ui/core/Dialog';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import MuiDialogContent from '@material-ui/core/DialogContent';
import MuiDialogActions from '@material-ui/core/DialogActions';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import AssessmentIcon from '@material-ui/icons/Assessment';
import Typography from '@material-ui/core/Typography';
import { getAllCommentsForTweet, getHiders } from './Module';

const styles = (theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(2),
  },
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
});

const DialogTitle = withStyles(styles)((props) => {
  const {
    children, classes, onClose,
  } = props;
  return (
    <MuiDialogTitle disableTypography className={classes.root}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButton
          aria-label="close"
          className={classes.closeButton}
          onClick={onClose}
        >
          <CloseIcon />
        </IconButton>
      ) : null}
    </MuiDialogTitle>
  );
});

const DialogContent = withStyles((theme) => ({
  root: {
    padding: theme.spacing(2),
  },
}))(MuiDialogContent);

const DialogActions = withStyles((theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(1),
  },
}))(MuiDialogActions);

export default function DialogAnalytics({ tweetId }) {
  const [hiders, setHiders] = useState([]);
  const [commentors, setCommentors] = useState([]);
  const [open, setOpen] = React.useState(false);
  const getData = async () => {
    await getAllCommentsForTweet(tweetId).then((res) => {
      const { comments } = res.data;
      setCommentors(comments.map((com) => com.user));
    }).catch((err) => console.log(err.message));
    await getHiders(tweetId).then((res) => {
      setHiders(res.data.hiders.map((hider) => hider.user));
    }).catch((err) => console.log(err.message));
  };
  const handleClickOpen = () => {
    setOpen(true);
  };
  const handleClose = () => {
    setOpen(false);
  };

  useEffect(() => {
    getData();
  }, []);
  useEffect(() => {

  }, [hiders, commentors, open]);
  return (
    <div>
      <div>
        <IconButton
          onClick={handleClickOpen}
          style={{
            borderRadius: '50%',
            padding: '3',
          }}
        >
          <AssessmentIcon />
        </IconButton>
        <span id="analytics_btn" style={{ color: '#0C8367' }}>Analytics</span>
      </div>
      <Dialog
        onClose={handleClose}
        aria-labelledby="customized-dialog-title"
        open={open}
      >
        <DialogTitle id="customized-dialog-title" onClose={handleClose}>
          Tweet analytics
        </DialogTitle>
        <DialogContent dividers>
          <div style={{ display: 'flex', minWidth: '100px' }}>
            <div>
              <span style={{ margin: 'auto', color: '#0C8367', fontWeight: 'bold' }}>Commentors</span>
              <ol style={{ margin: 'auto', listStyleType: 'circle' }}>
                {commentors.map((e) => (
                  <li>
                    {e}
                  </li>
                ))}
              </ol>
            </div>
            <div style={{ marginLeft: '5px', borderRadius: '8px', backgroundColor: '#D0EAE4' }} className="hiders">
              <div style={{ display: 'flex' }}>
                <span style={{ margin: 'auto', color: '#0C8367', fontWeight: 'bold' }}>Hiders</span>
              </div>

              <ol style={{ margin: 'auto', listStyleType: 'circle' }}>
                {hiders.map((hider) => (
                  <li key={hider}>
                    {hider}
                    {' '}
                  </li>
                ))}
              </ol>
            </div>
          </div>

        </DialogContent>
        <DialogActions />
      </Dialog>
    </div>
  );
}
