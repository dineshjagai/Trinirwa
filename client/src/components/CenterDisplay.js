import React, { useState, useEffect, useContext } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import PostAddIcon from '@material-ui/icons/PostAdd';
import LiveTvIcon from '@material-ui/icons/LiveTv';
import AddAPhotoIcon from '@material-ui/icons/AddAPhoto';
import TextField from '@material-ui/core/TextField';
import Tweet from './Tweet';
import idContext from './Context';
import { addTweet, deleteTweet } from './Module';
import './CommentBox.css';
// import Divider from '@material-ui/core/Divider';

const useStyles = makeStyles((theme) => ({
  button: {
    margin: theme.spacing(1),
  },
}));
const hash = require('object-hash');

export default function Displayer() {
  const [items, setItems] = useState(new Map());
  const [update, setUpdate] = useState(false);
  const [count, setCount] = useState(255);
  const [toDisplay] = useState(new Set());
  const classes = useStyles();
  const handleChange = (e) => {
    if ((e.target.value).length >= 0) {
      const u = count - 1;
      setCount(u);
    }
  };
  useEffect(() => {
    console.log('Im here I refreshed');
    setUpdate(false);
  }, [update]);
  const user = useContext(idContext);
  const postTweet = () => {
    setCount(255);
    const input = document.getElementById('tweet').value;
    const dateTime = new Date().toISOString();
    const tweetId = hash(`${input}${user}${dateTime}`);
    const newTweet = {
      username: user,
      tweet_id: tweetId,
      type: 'text',
      content: input,
      tweet_date: dateTime,
      tweet_likes: 0,
    };
    console.log(newTweet);
    const handleHideOrDelete = (id, isOwner) => {
      console.log(id);
      if (isOwner) {
        deleteTweet(id).then((res) => {
          console.log(res.message);
          setUpdate(true);
          toDisplay.delete(items.get(id));
          items.delete(id);
        }).catch((err) => {
          console.log(err.message);
        });
      }
    };
    const toAdd = <div className="tContainer"><Tweet handleDelete={handleHideOrDelete} data={newTweet} /></div>;
    const newItems = items;
    toDisplay.add(toAdd);
    console.log(toDisplay.length);
    newItems.set(tweetId, toAdd);
    setUpdate(true);
    setItems(newItems);
    console.log(items.length);
    addTweet(newTweet).then((res) => {
      console.log(res.message);
    }).catch((err) => {
      console.log(err.message);
    });
  };

  return (
    <div className="container_center">
      <div
        style={{
          backgroundColor: 'white',
          height: 'fit-content',
          borderRadius: '8px',
        }}
        className="whatIsOn"
      >
        <div
          style={{
            width: '90%',
            paddingTop: '4px',
            margin: 'auto',
          }}
        >
          <TextField
            multiline
            id="tweet"
            label="what's on your mind!!"
            variant="outlined"
            helperText={`${count} remaining chars`}
            inputProps={{ maxLength: 255 }}
            onChange={(e) => handleChange(e)}
            fullWidth
          />
        </div>
        <div
          style={{
            backgroundColor: 'white',
            borderRadius: '8px',
          }}
          className="buttons"
        >
          <Button
            variant="contained"
            color="secondary"
            className={classes.button}
            startIcon={<LiveTvIcon />}
          >
            Go live
          </Button>

          <Button
            variant="contained"
            color="secondary"
            className={classes.button}
            startIcon={<AddAPhotoIcon />}
          >
            Photo/video
          </Button>

          <Button
            variant="contained"
            color="secondary"
            className={classes.button}
            startIcon={<PostAddIcon />}
            onClick={postTweet}
          >
            Create post
          </Button>
        </div>
      </div>
      <div>
        {toDisplay}
      </div>
    </div>
  );
}
