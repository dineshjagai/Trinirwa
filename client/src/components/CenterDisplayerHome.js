import React, { useState, useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import PostAddIcon from '@material-ui/icons/PostAdd';
import LiveTvIcon from '@material-ui/icons/LiveTv';
import AddAPhotoIcon from '@material-ui/icons/AddAPhoto';
import TextField from '@material-ui/core/TextField';
import Tweet from './Tweet';
import { getCurrentUsername } from '../auth/authServices';

import {
  addTweet,
  deleteTweet,
  fetchAllTweets,
  hideTweet,
} from './Module';
import './CenterDisplay.css';
// import Divider from '@material-ui/core/Divider';

const useStyles = makeStyles((theme) => ({
  button: {
    margin: theme.spacing(1),
  },
}));
// eslint-disable-next-line import/no-unresolved
const hash = require('object-hash');

export default function DisplayerTweets() {
  const [items, setItems] = useState(new Map());
  const [update, setUpdate] = useState(false);
  const [count, setCount] = useState(255);
  const [toDisplay, setToDisplay] = useState(new Set());
  const classes = useStyles();
  const user = getCurrentUsername();
  const handleChange = (e) => {
    if ((e.target.value).length >= 0) {
      const u = 255 - (e.target.value).length;
      setCount(u);
    }
  };
  const handleHideOrDelete = (id, isOwner) => {
    if (isOwner) {
      deleteTweet(id).then((res) => {
        console.log('message: delete:', res.message);
        setUpdate(true);
        console.log('deleted:', toDisplay.delete(items.get(id)));
        items.delete(id);
      }).catch((err) => {
        console.log(err.message);
      });
    } else {
      hideTweet(id, user).then((res) => {
        console.log(res.status);
      }).catch((err) => console.log(err));
    }
  };

  useEffect(() => {
    setUpdate(false);
  }, [update]);
  const postTweet = () => {
    setCount(255);
    const input = document.getElementById('tweet').value;
    if (input.length === 0) return;
    const dateTime = new Date().toISOString();
    const tweetId = hash(`${input}${user}${dateTime}`);
    const newTweet = {
      user,
      tweet_id: tweetId,
      type: 'text',
      content: input,
      tweet_date: dateTime,
      tweet_likes: 0,
    };
    const toAdd = <div className="tContainer"><Tweet handleDelete={handleHideOrDelete} data={newTweet} /></div>;
    const newItems = items;
    const newToDisplay = new Set([toAdd, ...toDisplay]);
    setToDisplay(newToDisplay);
    newItems.set(tweetId, toAdd);
    setUpdate(true);
    setItems(newItems);
    console.log(items.length);
    addTweet(newTweet).then((res) => {
      console.log(res.message);
    }).catch((err) => {
      console.log(err.message);
    });
  };

  const getData = async () => {
    fetchAllTweets(user).then((res) => {
      const { tweets } = res.data;
      tweets.forEach((e) => {
        const toAdd = <div className="tContainer"><Tweet handleDelete={handleHideOrDelete} data={e} /></div>;
        toDisplay.add(toAdd);
        items.set(e.tweet_id, toAdd);
      });
      setUpdate(true);
    }).catch((err) => console.log(err));
  };
  useEffect(() => {
    getData();
  }, []);
  return (
    <div className="container_center">
      <div
        style={{
          backgroundColor: 'white',
          height: 'fit-content',
          borderRadius: '8px',
        }}
        className="whatIsOn"
      >
        <div
          style={{
            width: '90%',
            paddingTop: '4px',
            margin: 'auto',
          }}
        >
          <TextField
            multiline
            id="tweet"
            label="what's on your mind!!"
            variant="outlined"
            helperText={`${count} remaining chars`}
            inputProps={{ maxLength: 255 }}
            onChange={(e) => handleChange(e)}
            fullWidth
          />
        </div>
        <div
          style={{
            backgroundColor: 'white',
            borderRadius: '8px',
          }}
          className="buttons"
        >
          <Button
            variant="contained"
            color="secondary"
            className={classes.button}
            startIcon={<LiveTvIcon />}
          >
            Go live
          </Button>

          <Button
            variant="contained"
            color="secondary"
            className={classes.button}
            startIcon={<AddAPhotoIcon />}
          >
            Photo/video
          </Button>

          <Button
            variant="contained"
            color="secondary"
            className={classes.button}
            startIcon={<PostAddIcon />}
            onClick={postTweet}
          >
            Create post
          </Button>
        </div>
      </div>
      <div>
        {toDisplay}
      </div>
    </div>
  );
}
