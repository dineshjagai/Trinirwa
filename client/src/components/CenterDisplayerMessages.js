/* eslint-disable no-unused-vars */
/* eslint-disable max-len */
import React, { useState, useEffect } from 'react';
// import { useHistory } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import PostAddIcon from '@material-ui/icons/PostAdd';
import AddAPhotoIcon from '@material-ui/icons/AddAPhoto';
import VideoLibraryIcon from '@material-ui/icons/VideoLibrary';
import LibraryMusicIcon from '@material-ui/icons/LibraryMusic';
import TextField from '@material-ui/core/TextField';
import Message from './Message';
import Tweet from './Tweet';
// eslint-disable-next-line no-unused-vars
import { getCurrentUsername, getCurrentReceiver } from '../auth/authServices';
import './CenterDisplayerHome.css';

import {
  addMessage,
  fetchMessages,
} from './Module';
import './CenterDisplay.css';

// import Divider from '@material-ui/core/Divider';

const useStyles = makeStyles((theme) => ({
  button: {
    margin: theme.spacing(1),
  },
}));
// eslint-disable-next-line import/no-unresolved
const hash = require('object-hash');

export default function DisplayerMessages() {
  const receiver = getCurrentReceiver();
  console.log(`receiver = ${receiver}`);
  //   const history = useHistory();

  const [items] = useState(new Map());
  const [update, setUpdate] = useState(false);
  const [count, setCount] = useState(255);
  const [toDisplay] = useState(new Set());
  const classes = useStyles();
  const user = getCurrentUsername();
  const handleChange = (e) => {
    if ((e.target.value).length >= 0) {
      const u = 255 - (e.target.value).length;
      setCount(u);
    }
  };
  useEffect(() => {
    console.log('refreshed');
    setUpdate(false);
  }, [update]);
  const postMessage = () => {
    setCount(255);
    const input = document.getElementById('message').value;
    if (typeof input === 'undefined') {
      return;
    }
    if (input.length === 0) return;
    const dateTime = new Date().toISOString();
    const messageId = hash(`${input}${user}${dateTime}`);
    const newMessage = {
      user,
      message_id: messageId,
      type: 'text',
      content: input,
      message_date: dateTime,
      receiver,
    };
    // const toAdd = <div className="tContainer"><Tweet handleDelete={handleHideOrDelete} data={newMessage} /></div>;
    // // const newItems = items;
    // const newToDisplay = new Set([toAdd, ...toDisplay]);
    // const newItems = items;

    // setToDisplay(newToDisplay);
    // items.set(messageId, toAdd);
    // newItems.set(messageId, toAdd);
    // setUpdate(true);
    // setItems(newItems);
    // console.log('items length', items.length);
    addMessage(newMessage).then((res) => {
      console.log(res.message);
    //   window.location.reload();
    }).catch((err) => {
      console.log(err.message);
    });
  };

  const getData = async () => {
    fetchMessages(user, receiver).then((res) => {
      const { messages } = res.data;
      console.log(`ttt--${messages}`);
      messages.forEach((e) => {
        const toAdd = <div className="tContainer"><Message data={e} /></div>;
        toDisplay.add(toAdd);
        items.set(e.tweet_id, toAdd);
      });
      setUpdate(true);
    }).catch((err) => console.log(err));
  };
  useEffect(() => {
    getData();
  }, []);

  const postPicture = (e) => {
    // upload picture
    const formdata = new FormData();
    // console.log(e.target.files);
    const fakePath = document.getElementById('fileInputPicture').value;
    formdata.append('image', e.target.files[0], fakePath);

    const requestOptions = {
      method: 'POST',
      body: formdata,
      redirect: 'follow',
    };
    const input = document.getElementById('message').value;
    if (typeof input === 'undefined') {
      return;
    }
    const dateTime = new Date().toISOString();
    const messageId = hash(`${input}${user}${dateTime}`);

    fetch('/api/uploadFile', requestOptions)
      .then((response) => response.text())
      .then((result) => {
        // console.log(`---uploading a pic ---:${result}`);
        const newMessage = {
          user,
          message_id: messageId,
          type: 'picture',
          content: JSON.parse(result).data,
          message_date: dateTime,
          receiver,
        };
        console.log(`is this good? ${newMessage}`);
        // const toAdd = <div className="tContainer"><Tweet handleDelete={handleHideOrDelete} data={newMessage} /></div>;
        // const newItems = items;
        // const newToDisplay = new Set([toAdd, ...toDisplay]);
        // setToDisplay(newToDisplay);
        // newItems.set(messageId, toAdd);
        // setUpdate(true);
        // setItems(newItems);
        // console.log(items.length);
        addMessage(newMessage).then((res) => {
          console.log(res.message);
        //   window.location.reload();
        }).catch((err) => {
          console.log(err.message);
        });
      })
      .catch((error) => {
        console.log(error);
      });
  };

  const postVideo = (e) => {
    // upload video
    const formdata = new FormData();
    // console.log(e.target.files);
    const fakePath = document.getElementById('fileInputVideo').value;
    formdata.append('image', e.target.files[0], fakePath);

    const requestOptions = {
      method: 'POST',
      body: formdata,
      redirect: 'follow',
    };
    const input = document.getElementById('message').value;
    if (typeof input === 'undefined') {
      return;
    }
    const dateTime = new Date().toISOString();
    const messageId = hash(`${input}${user}${dateTime}`);

    fetch('/api/uploadFile', requestOptions)
      .then((response) => response.text())
      .then((result) => {
        const newMessage = {
          user,
          message_id: messageId,
          type: 'video',
          content: JSON.parse(result).data,
          message_date: dateTime,
          receiver,
        };
        // console.log(`is this good? ${newMessage.user}`);
        // const toAdd = <div className="tContainer"><Tweet handleDelete={handleHideOrDelete} data={newMessage} /></div>;
        // const newItems = items;
        // const newToDisplay = new Set([toAdd, ...toDisplay]);
        // setToDisplay(newToDisplay);
        // newItems.set(messageId, toAdd);
        // setUpdate(true);
        // setItems(newItems);
        // console.log(items.length);
        addMessage(newMessage).then((res) => {
          console.log(res.message);
        //   window.location.reload();
        }).catch((err) => {
          console.log(err.message);
        });
      })
      .catch((error) => {
        console.log(error);
      });
  };

  const postSong = (e) => {
    // upload song
    const formdata = new FormData();
    // console.log(e.target.files);
    const fakePath = document.getElementById('fileInputSong').value;
    formdata.append('image', e.target.files[0], fakePath);

    const requestOptions = {
      method: 'POST',
      body: formdata,
      redirect: 'follow',
    };
    const input = document.getElementById('message').value;
    if (typeof input === 'undefined') {
      return;
    }
    const dateTime = new Date().toISOString();
    const messageId = hash(`${input}${user}${dateTime}`);

    fetch('/api/uploadFile', requestOptions)
      .then((response) => response.text())
      .then((result) => {
        const newMessage = {
          user,
          message_id: messageId,
          type: 'song',
          content: JSON.parse(result).data,
          message_date: dateTime,
          receiver,
        };
        // console.log(`is this good? ${newMessage.user}`);
        // const toAdd = <div className="tContainer"><Tweet handleDelete={handleHideOrDelete} data={newMessage} /></div>;
        // const newItems = items;
        // const newToDisplay = new Set([toAdd, ...toDisplay]);
        // setToDisplay(newToDisplay);
        // newItems.set(messageId, toAdd);
        // setUpdate(true);
        // setItems(newItems);
        // console.log(items.length);
        addMessage(newMessage).then((res) => {
        //   window.location.reload();
          console.log(res.message);
        }).catch((err) => {
          console.log(err.message);
        });
      })
      .catch((error) => {
        console.log(error);
      });
  };

  const postPictureTrigger = () => {
    document.getElementById('fileInputPicture').click();
  };

  const postVideoTrigger = () => {
    document.getElementById('fileInputVideo').click();
  };

  const postSongTrigger = () => {
    document.getElementById('fileInputSong').click();
  };
  console.log('td', toDisplay);
  return (
    <div className="container_center">
      <div
        style={{
          backgroundColor: 'white',
          height: 'fit-content',
          borderRadius: '8px',
        }}
        className="whatIsOn"
      >
        <div
          style={{
            width: '90%',
            paddingTop: '4px',
            margin: 'auto',
          }}
        >
          <TextField
            multiline
            id="message"
            label="what's on your mind!!"
            variant="outlined"
            helperText={`${count} remaining chars`}
            inputProps={{ maxLength: 255 }}
            onChange={(e) => handleChange(e)}
            fullWidth
          />
        </div>

        <div
          style={{
            backgroundColor: 'white',
            borderRadius: '8px',
          }}
          className="buttons"
        >

          <Button
            variant="contained"
            color="secondary"
            id="button-media"
            className={classes.button}
            startIcon={<AddAPhotoIcon />}
            onClick={postPictureTrigger}
          >
            Photo
          </Button>
          <Button
            variant="contained"
            color="secondary"
            id="button-media"
            className={classes.button}
            startIcon={<VideoLibraryIcon />}
            onClick={postVideoTrigger}
          >
            Video
          </Button>
          <Button
            variant="contained"
            color="secondary"
            id="button-media"
            className={classes.button}
            startIcon={<LibraryMusicIcon />}
            onClick={postSongTrigger}
          >
            Song
          </Button>

          <Button
            variant="contained"
            color="secondary"
            className={classes.button}
            startIcon={<PostAddIcon />}
            onClick={postMessage}
          >
            Create post
          </Button>
          <input
            className="FileUpload"
            accept=".jpg,.png,.gif"
            id="fileInputPicture"
            type="file"
            onChange={postPicture}
            hidden
          />
          <input
            className="FileUpload"
            id="fileInputVideo"
            type="file"
            onChange={postVideo}
            hidden
          />
          <input
            className="FileUpload"
            id="fileInputSong"
            type="file"
            onChange={postSong}
            hidden
          />
        </div>
      </div>
      <div>
        {toDisplay}
      </div>
    </div>
  );
}
