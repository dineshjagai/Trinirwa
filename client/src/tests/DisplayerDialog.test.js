import React from 'react';
import renderer from 'react-test-renderer';
import ScrollDialog from '../components/DisplayerDialog';
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom';
import { act } from 'react-dom/test-utils';
import { getFollowers } from '../components/Module';


test('Test DisplayerDialog snapshot', () => {
  const component = renderer.create(<ScrollDialog 
    title={'Title'}
    secondTitle={'title'}
  />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Test DisplayerDialog Close content', async () => {
  const { getByText } = render(<ScrollDialog 
    title={'Title'}
    secondTitle={'second Title'}
    getFunction={getFollowers}
  />);
  const titleBtn = screen.getByText('Title', { selector: 'button' });
  expect(titleBtn).toBeInTheDocument();

  await act(async () => {
    await fireEvent.click(titleBtn);   
  });
  const closeBtn = screen.getByRole('button', { name : 'Close'});
  expect(screen.getByText('second Title')).toBeInTheDocument();
  await act(async () => {
    await fireEvent.click(closeBtn);   
  });
});
