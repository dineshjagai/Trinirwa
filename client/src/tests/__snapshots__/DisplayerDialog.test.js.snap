// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DisplayerDialog snapshot 1`] = `
<div>
  <button
    className="btn"
    id="delete"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "inherit",
        "border": "none",
        "color": "#00695c",
      }
    }
    type="button"
  >
    Title
  </button>
</div>
`;
